ospd/scan.py

```diff
@@ -347,38 +347,44 @@ def get_host_count(self, scan_id: str) -> int:
 
         return total_hosts
 
-    def get_ports(self, scan_id: str):
+    def get_ports(self, scan_id: str) -> str:
         """ Get a scan's ports list.
         """
         target = self.scans_table[scan_id].get('target')
         ports = target.pop('ports')
         self.scans_table[scan_id]['target'] = target
         return ports
 
-    def get_exclude_hosts(self, scan_id: str):
+    def get_exclude_hosts(self, scan_id: str) -> str:
         """ Get an exclude host list for a given target.
         """
         return self.scans_table[scan_id]['target'].get('exclude_hosts')
 
-    def get_finished_hosts(self, scan_id: str):
+    def get_finished_hosts(self, scan_id: str) -> str:
         """ Get the finished host list sent by the client for a given target.
         """
         return self.scans_table[scan_id]['target'].get('finished_hosts')
 
-    def get_credentials(self, scan_id: str):
+    def get_credentials(self, scan_id: str) -> Dict[str, Dict[str, str]]:
         """ Get a scan's credential list. It return dictionary with
         the corresponding credential for a given target.
         """
         return self.scans_table[scan_id]['target'].get('credentials')
 
-    def get_target_options(self, scan_id: str):
+    def get_target_options(self, scan_id: str) -> Dict[str, str]:
         """ Get a scan's target option dictionary.
         It return dictionary with the corresponding options for
         a given target.
         """
         return self.scans_table[scan_id]['target'].get('options')
 
-    def get_vts(self, scan_id: str) -> Dict:
+    def get_target(
+        self, scan_id: str
+    ) -> Dict[str, Union[str, Dict[str, Union[str, Dict[str, str]]]]]:
```

Comment:
Holy crap :flushed:

Reply:
All target information is there. Credentials (for different services), target options (alive tests method and reverse_lookup options), port list, hosts (to be scanned, finished and excluded).

I removed this method, because is not necessary anymore. I have just simplified the start_scan() method.